### ISSUE TITLE
bug: TypeError if no device exists. (run android)

### ISSUE THREAD

**Issue by twoco** [ionic-team/capacitor#6056]
```none
> ionic capacitor run android

? Which device would you like to target? (Use arrow keys)
TypeError: Cannot read properties of undefined (reading 'value')
```

We could show "No device found". instead of accessing an undefined variable (error).

cap v3.4.3, Windows 10

**Comment by jcesarmobile** [comment]
I can't reproduce.
`ionic capacitor run android` will run `npx cap run android` internally, which will then call `npx native-run android --list --json` to get the list of available emulators and devices.
The thing is `npx native-run android --list --json` will create a new emulator if there aren't any. 
Might be there when the "cannot read properties of undefined" happens.

Can you provide your native-run and @ionic/cli versions?

```
ionic --version
npx native-run --version
```
and then run `npx native-run android --list --json` and see if it returns at least an emulator?

**Comment by twoco** [comment]
I run ionic as npm script. So my workspace version is used. It's 6.19.0 (same with 6.20.3). npx native-run is 1.7.0. 

Script: `ionic capacitor run android`

Ok as I see now the error only appears then hit enter when no list appears. 

> ? Which device would you like to target? 
> _
> _enter_
> TypeError: Cannot read properties of undefined (reading 'value')

Without knowing the code, I think there is no if-condition or optional chaning of the selected entry. The selected item variable is just used direactly and tries to read ts value. 

```
$ npx native-run android --list --json
```

**without any device/emulator connected**
```json
{
  "devices": [],
  "virtualDevices": [],
  "errors": [
    {
      "error": "No suitable API installation found. Use --sdk-info to reveal missing packages and other issues.",
      "code": "ERR_UNSUITABLE_API_INSTALLATION"
    }
  ]
}
```

**device connected**

```json
{
  "devices": [
    {
      "platform": "android",
      "model": "BLABLABLA",
      "sdkVersion": "28",
      "id": "BLABLABLA"
    }
  ],
  "virtualDevices": [],
  "errors": [
    {
      "error": "No suitable API installation found. Use --sdk-info to reveal missing packages and other issues.",
      "code": "ERR_UNSUITABLE_API_INSTALLATION"
    }
  ]
}
```

Weird error. I already tried `--sdk-info` but it seems that this arg does not exist.

I wonder what library calls `value` of the selected entry (in this case `undefined`).
But anyway, I would not show "Which device would you like to target?" if the list is empty. Or show "[No device found]" below.

Another issue is to automatically select the connected device. Espacially if only one device is connected. In my list I only have one device. And sometimes I run this command and forget to select it. ^^ But I remember the early Capacitor days where you needed to run the project in Android Studio. Glad to see that Capacitor works over CLI now (like Cordova) to run on device. 👍 

Anyway this issue described here is not critical. I just thought I could report this minor issue to improve it a bit.

Btw. I don't understand why this command "will create a new emulator". It would also be wrong behavior to do so.

Long story short: We just need to know where the `value` is accessed and simply check nullish (undefined) here.

**Comment by jcesarmobile** [comment]
The problem is your android SDK is not correctly installed, it’s missing some packages.
Capacitor CLI should forward the error from native-run instead of showing the empty list.
https://github.com/ionic-team/native-run/wiki/Android-Errors#err_unsuitable_api_installation

the --sdk-info parameter is for native-run command, it doesn’t exist on capacitor nor ionic

**Comment by twoco** [comment]
To be honest, I have no issues. I can select my device and build my app without any issue. But I'll take a look to the missing packages. It seems that the latest API are checked. But nothing critical I see. ... Maybe Google Play Store Console will annoy me when uploading the app. I'll take care then. ^^ But it seems that "Google Play Intel x86 Atom System Image" does not really matter for me. Whatever...

`npx native-run android --sdk-info`

<details>
  <summary>View result (click me)</summary>
  
  ```
SDK Location:         C:\Users\User\AppData\Local\Android\Sdk
AVD Home:             C:\Users\User\.android\avd

API Level:            32
Packages:             Android SDK Platform 32  platforms;android-32  1
(!) Missing Packages: Google Play Intel x86 Atom System Image  system-images;android-32;google_apis_playstore;x86  /.+/


API Level:            31
Packages:             Android SDK Platform 31  platforms;android-31  1
(!) Missing Packages: Google Play Intel x86 Atom System Image  system-images;android-31;google_apis_playstore;x86  /.+/


API Level:            30
Packages:             Android SDK Platform 30  platforms;android-30  3
(!) Missing Packages: Google Play Intel x86 Atom System Image  system-images;android-30;google_apis_playstore;x86  /.+/


API Level:            29
Packages:             Android SDK Platform 29  platforms;android-29  5
                      Sources for Android 29  sources;android-29  1
(!) Missing Packages: Google Play Intel x86 Atom System Image  system-images;android-29;google_apis_playstore;x86  /.+/


API Level:            28
Packages:             Android SDK Platform 28  platforms;android-28  6
(!) Missing Packages: Google Play Intel x86 Atom System Image  system-images;android-28;google_apis_playstore;x86  /.+/


API Level:            25
Packages:             Google APIs Intel x86 Atom System Image  system-images;android-25;google_apis;x86  17
(!) Missing Packages: Android SDK Platform 25  platforms;android-25


Tools:

Android Emulator  emulator  30.8.4
Android SDK Build-Tools 28.0.3  build-tools;28.0.3  28.0.3
Android SDK Build-Tools 29.0.2  build-tools;29.0.2  29.0.2
Android SDK Build-Tools 29.0.3  build-tools;29.0.3  29.0.3
Android SDK Build-Tools 30.0.2  build-tools;30.0.2  30.0.2
Android SDK Build-Tools 30.0.3  build-tools;30.0.3  30.0.3
Android SDK Platform-Tools  platform-tools  31.0.3
Android SDK Tools  tools  26.1.1
SDK Patch Applier v4  patcher;v4  1
  ```
</details>
